### DISCUSSION TITLE
The maximum distance that proximity sensor IR can calculate

### DISCUSSION THREAD

**Discussion by ali1hammoud** [cyberbotics/webots#6409] [General]
What is the maximum distance that proximity sensor IR can calculate?
according to the doc, we can change the distance in Lookup Table, I used E-puck robot, converted it to base node, and changed the values in lookup table as:
[[0.0e+00 3.0e+03 0.0e+00]
 [1.0e-01 2.0e+03 0.0e+00]
 [2.5e-01 1.0e+03 0.0e+00]
 [1.0e+00 0.0e+00 0.0e+00]]
So, the sensor will return 0 for 1 m and 3000 for 0 m.
I changed the redColorSensitivity to zero.

Then I started the simulation and got the values and found that the sensor will give me 0.0 from distance 0.417 m to 1 m, and 772 in 0.416.
The distance 0.416 is the maximum distance that the sensor can reach and sense, I searched to find the function in code that responsible to calculate the distance in simulation, but I couldn't find it, I searched in doc about the equation that represents the value of sensor reading, I couldn't find it either, the only thing I found the reflection factor as follows: f = 0.2 + 0.8 * red_level * (1 - 0.5 * roughness) * (1 - 0.5 * occlusion).

My question is 0.416 is the maximum distance can IR sensor sense? if not, what is the maximum distance, and what is the equation or function in code that return the reading value?

**Comment by omichel** [comment]
There is no maximum value to the range of a IR distance sensor, as you can define it in the [lookup table](https://cyberbotics.com/doc/reference/distancesensor#lookup-table). The source code computing the return value of a distance sensor is located [here](https://github.com/cyberbotics/webots/blob/d76e17aaf2610d50b71b23cfcea168f824387942/src/webots/nodes/WbDistanceSensor.cpp#L491).

**Reply by ali1hammoud** [reply]
Thank you for your answer.
I will test it again and check the code, maybe I made a mistake, I have already double checked the world and sensor on e-puck, I think that I will test the sensor alone.
I will write again for results.

**Comment by ali1hammoud** [comment]
After testing the distance sensor (IR type) alone without e-puck robot, I got the distance value even if it more than 2 meters.
I will try to find the problem with E-puck robot.

**Comment by ali1hammoud** [comment]
I found the problem.
The solid (Parent Node of sensor) has a small rotation and the sensor also has a small rotation, according to that the ray of sensor hit the obstacles at 0.4 meters and went over them if the distance more, the height of obstacles is 0.01 meters.

Thank you for helping.
